pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go

```diff
@@ -465,6 +485,37 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
 	return nil
 }
 
+func skipCleanupManagedClusterResources(klusterlet *operatorapiv1.Klusterlet, mode operatorapiv1.InstallMode) bool {
```

Comment:
add a comments here
1. we skip only in hosted mode and there is not hosted finalizer on klusterelt. The scenario happens when we cannot build managedclusterclient in hosted mode.